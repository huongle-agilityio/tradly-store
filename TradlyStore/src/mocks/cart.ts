import { Cart } from '@/interfaces';

export const CART_QUANTITY = [
  {
    value: '1',
    label: '1',
  },
  {
    value: '2',
    label: '2',
  },
  {
    value: '3',
    label: '3',
  },
  {
    value: '4',
    label: '4',
  },
  {
    value: '5',
    label: '5',
  },
  {
    value: '6',
    label: '6',
  },
  {
    value: '7',
    label: '7',
  },
  {
    value: '8',
    label: '8',
  },
  {
    value: '9',
    label: '9',
  },
  {
    value: '10',
    label: '10',
  },
];

export const CART_lIST: Cart[] = [
  { id: '1', name: 'Item 1', quantity: 2, price: 100, image: '' },
  { id: '2', name: 'Item 2', quantity: 1, price: 50, image: '' },
];
