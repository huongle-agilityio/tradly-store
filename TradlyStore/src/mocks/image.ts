export const IMAGES = 'https://picsum.photos/seed/696/3000/2000';

export const LIST_IMAGES = Array.from({ length: 4 }, () => IMAGES);
