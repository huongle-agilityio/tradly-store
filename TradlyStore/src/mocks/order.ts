// Interfaces
import { OderTracking } from '@/interfaces';

export const ORDER_TRACKING: OderTracking[] = [
  {
    title: 'Order Placed',
    description: 'Order#123455 from Fashion Point',
    date: '05/08/2019',
    time: '11:10 AM',
  },
  {
    title: 'Payment Confirmed',
    description: 'Payment Confirmed Status',
    date: '05/08/2019',
    time: '11:10 AM',
    isActive: true,
  },
  {
    title: 'Processed',
    description: 'Processed Status',
    date: '05/08/2019',
    time: '11:10 AM',
  },
  {
    title: 'Delivered',
    description: 'Delivered Status',
    date: '05/08/2019',
    time: '11:10 AM',
  },
];
