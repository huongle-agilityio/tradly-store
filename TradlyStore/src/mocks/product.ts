export const INT_PRODUCT_CARD = {
  title: '',
  price: 0,
  image: '',
  discount: 0,
  store: {
    name: '',
    image: '',
  },
};

export const ADDITIONAL_DETAILS = [
  {
    label: 'Shipping',
    value: 'shipping',
  },
  {
    label: 'Available',
    value: 'available',
  },
  {
    label: 'Cash on delivery',
    value: 'cashOnDelivery',
  },
];

export const PRODUCT_TEST = [
  {
    id: 1,
    documentId: '1',
    title: 'Grape Soda',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    slideImages: [
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    ],
    quantity: 50,
    price: 34,
    discount: 50,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 2,
    documentId: '1',
    title: 'Tropical Punch',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image:
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    slideImages: [
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    ],
    quantity: 22,
    price: 48,
    discount: 39,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 3,
    documentId: '1',
    title: 'Pineapple Punch',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    slideImages: [
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
      'https://i.ibb.co/jvGv5WLk/Boba-Cat-Taro-Bubble-Tea-Drink-315ml.jpg',
    ],
    quantity: 20,
    price: 26,
    discount: 29,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 4,
    documentId: '1',
    title: 'Peach Tea',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image:
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    slideImages: [
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
      'https://i.ibb.co/5xRKFrht/thung-24-chai-nuoc-ngot-coca-cola-390ml.png',
    ],
    quantity: 21,
    price: 32,
    discount: 48,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 5,
    documentId: '1',
    title: 'Berry Blast',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    slideImages: [
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    ],
    quantity: 13,
    price: 25,
    discount: 28,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 6,
    documentId: '1',
    title: 'Lemonade',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    slideImages: [
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    ],
    quantity: 19,
    price: 30,
    discount: 27,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 7,
    documentId: '1',
    title: 'Watermelon Soda',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    slideImages: [
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
      'https://i.ibb.co/DgwdzQb7/DONGSUH-Buckwheat-Tea-1-5gr.jpg',
    ],
    quantity: 49,
    price: 40,
    discount: 16,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
  {
    id: 8,
    documentId: '1',
    title: 'Mango Juice',
    description:
      "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s.",
    image: 'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
    slideImages: [
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
      'https://i.ibb.co/qMntFSWn/Crush-Peach-Soda-355ml.jpg',
    ],
    quantity: 33,
    price: 28,
    discount: 12,
    priceType: 'Fixed',
    location: 'Lorem Ipsum',
    category: 'beverages',
    additionalDetails: ['cashOnDelivery', 'available'],
    store: {
      id: 1,
      documentId: '1',
      image: 'string',
      coverImage: 'string',
      username: 'string',
      products: [],
    },
  },
];
