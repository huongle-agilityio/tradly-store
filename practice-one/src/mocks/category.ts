import { IMAGES } from '@/constants';

export const CATEGORIES = [
  {
    id: 1,
    name: 'Beverages',
    value: 'beverages',
    image: IMAGES.CATEGORY_BEVERAGES,
  },
  {
    id: 2,
    name: 'Bread & Bakery',
    value: 'bread',
    image: IMAGES.CATEGORY_BREAD_BAKERY,
  },
  {
    id: 3,
    name: 'Vegetables',
    value: 'vegetables',
    image: IMAGES.CATEGORY_VEGETABLES,
  },
  {
    id: 4,
    name: 'Fruit',
    value: 'fruit',
    image: IMAGES.CATEGORY_FRUIT,
  },
  {
    id: 5,
    name: 'Egg',
    value: 'egg',
    image: IMAGES.CATEGORY_EGG,
  },
  {
    id: 6,
    name: 'Frozen veg',
    value: 'frozen-veg',
    image: IMAGES.CATEGORY_FROZEN_VEG,
  },
  {
    id: 7,
    name: 'Homecare',
    value: 'homecare',
    image: IMAGES.CATEGORY_HOME_CARE,
  },
  {
    id: 8,
    name: 'Pet Care',
    value: 'pet-care',
    image: IMAGES.CATEGORY_PET_CARE,
  },
];
