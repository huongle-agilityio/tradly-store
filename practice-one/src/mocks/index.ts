export * from './cart';
export * from './product';
export * from './category';
export * from './onboarding';
export * from './order';
export * from './user';
