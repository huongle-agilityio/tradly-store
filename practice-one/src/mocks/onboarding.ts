import { IMAGE_DETAILS } from '@/constants';

export const ONBOARDING_STEPS_IMAGES = [
  {
    image: IMAGE_DETAILS.ONLINE_SHOPPING.src,
    title: 'Empowering Artisans, Farmers & Micro Business',
    alt: IMAGE_DETAILS.ONLINE_SHOPPING.alt,
  },
  {
    image: IMAGE_DETAILS.DIGITAL_COLLABORATION.src,
    title: 'Connecting NGOs, Social Enterprises with Communities',
    alt: IMAGE_DETAILS.DIGITAL_COLLABORATION.alt,
  },
  {
    image: IMAGE_DETAILS.TECHNICAL_ANALYSIS.src,
    title: 'Donate, Invest & Support infrastructure projects',
    alt: IMAGE_DETAILS.TECHNICAL_ANALYSIS.alt,
  },
];
