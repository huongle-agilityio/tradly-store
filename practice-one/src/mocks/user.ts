import { User } from '@/interfaces';

export const USER: User = {
  id: '123',
  email: 'test@example.com',
  name: 'John Doe',
  avatar: '',
  username: 'johndoe',
  phone: '123456789',
};
