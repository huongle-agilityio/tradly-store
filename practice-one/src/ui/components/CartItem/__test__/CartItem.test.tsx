import { render, screen, fireEvent } from '@testing-library/react-native';

// Components
import { CartItem } from '..';

// Mocks
import { CART_QUANTITY } from '@/mocks';

const mockOnRemoveItem = jest.fn();
const mockOnUpdateQuantityItem = jest.fn();

const defaultProps = {
  id: '1',
  image: 'https://example.com/image.jpg',
  name: 'Test Product',
  quantity: 2,
  price: 100,
  discount: 10,
  onRemoveItem: mockOnRemoveItem,
  onUpdateQuantityItem: mockOnUpdateQuantityItem,
};

describe('CartItem', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Should renders CartItem correctly', () => {
    const { toJSON } = render(<CartItem {...defaultProps} />);

    expect(screen.getByText('Test Product')).toBeTruthy();
    expect(screen.getByText('$90')).toBeTruthy();
    expect(screen.getByText('$100')).toBeTruthy();
    expect(screen.getByText('10% off')).toBeTruthy();
    expect(screen.getByText('Remove')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('Should calls onRemoveItem when "Remove" button is pressed', () => {
    render(<CartItem {...defaultProps} />);

    fireEvent.press(screen.getByText('Remove'));

    expect(mockOnRemoveItem).toHaveBeenCalledTimes(1);
    expect(mockOnRemoveItem).toHaveBeenCalledWith('1');
  });

  it('Should calls onUpdateQuantityItem when quantity is changed', () => {
    render(<CartItem {...defaultProps} />);

    fireEvent.press(screen.getByTestId('quantity-select'));
    fireEvent.press(screen.getByText(CART_QUANTITY[1].label));

    expect(mockOnUpdateQuantityItem).toHaveBeenCalledTimes(1);
    expect(mockOnUpdateQuantityItem).toHaveBeenCalledWith('1', '2');
  });

  it('Should does not display discount if there is no discount', () => {
    render(<CartItem {...defaultProps} discount={undefined} />);

    expect(screen.getByText('$100')).toBeTruthy();
    expect(screen.queryByText('10% off')).toBeFalsy();
  });
});
