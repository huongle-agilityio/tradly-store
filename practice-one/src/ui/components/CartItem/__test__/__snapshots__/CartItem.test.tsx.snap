// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Should renders CartItem correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "gap": 15,
          "paddingHorizontal": 16,
        },
        {
          "paddingBottom": 12,
          "paddingTop": 30,
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoImage
      accessibilityLabel="image of Test Product product"
      borderRadius={10}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={120}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "height": 16,
            "uri": "blurhash:/%7CrF%3FhV%252WCj%5Bayj%5Ba%7Cj%5Baz_NaeWBj@ayfRayfQfQM%7BM%7Cazj%5Bazf6fQfQfQIpWXofj%5Bayj%5Bj%5BfQayWCoeoeaya%7Dj%5BayfQa%7BoLj%3Fj%5BWVj%5Bayayj%5BfQoff7azayj%5Bayj%5Bj%5Bayofayayayj%5BfQj%5Bayayj%5Bayfjj%5Bj%5Bayjuayj%5B",
            "width": 16,
          },
        ]
      }
      source={
        [
          {
            "uri": "https://example.com/image.jpg",
          },
        ]
      }
      style={
        {
          "borderRadius": 10,
          "height": 120,
          "width": 120,
        }
      }
      transition={
        {
          "duration": 1000,
        }
      }
      width={120}
    />
    <View
      style={
        {
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 14,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 16,
            },
            {
              "fontFamily": "Montserrat-Medium",
              "fontWeight": "500",
            },
            {
              "color": "#4f4f4f",
            },
            undefined,
          ]
        }
      >
        Test Product
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 18,
              },
              {
                "fontFamily": "Montserrat-SemiBold",
                "fontWeight": "700",
              },
              {
                "color": "#33907c",
              },
              undefined,
            ]
          }
        >
          $
          90
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "#4f4f4f",
                },
                {
                  "textDecorationLine": "line-through",
                },
              ]
            }
          >
            $
            100
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "#4f4f4f",
                },
                undefined,
              ]
            }
          >
            10
            % off
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12,
                "lineHeight": 21,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "#4f4f4f",
              },
              undefined,
            ]
          }
        >
          Qty:
        </Text>
        <View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="ios_touchable_wrapper"
            >
              <View
                pointerEvents="box-only"
              >
                <TextInput
                  editable={false}
                  style={
                    [
                      undefined,
                      {},
                    ]
                  }
                  testID="text_input"
                  value="2"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderTopColor": "#dbdbde",
        "borderTopWidth": 0.5,
        "opacity": 1,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 14,
            "lineHeight": 16,
          },
          {
            "fontFamily": "Montserrat-Medium",
            "fontWeight": "500",
          },
          {
            "color": "#4f4f4f",
          },
          undefined,
        ]
      }
    >
      Remove
    </Text>
  </View>
</View>
`;

exports[`CartItem Should renders CartItem correctly without actions 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
    }
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
          "gap": 15,
          "paddingHorizontal": 16,
        },
        {
          "paddingVertical": 25,
        },
      ]
    }
  >
    <ViewManagerAdapter_ExpoImage
      accessibilityLabel="image of Test Product product"
      borderRadius={10}
      containerViewRef={"[React.ref]"}
      contentFit="cover"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={120}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "height": 16,
            "uri": "blurhash:/%7CrF%3FhV%252WCj%5Bayj%5Ba%7Cj%5Baz_NaeWBj@ayfRayfQfQM%7BM%7Cazj%5Bazf6fQfQfQIpWXofj%5Bayj%5Bj%5BfQayWCoeoeaya%7Dj%5BayfQa%7BoLj%3Fj%5BWVj%5Bayayj%5BfQoff7azayj%5Bayj%5Bj%5Bayofayayayj%5BfQj%5Bayayj%5Bayfjj%5Bj%5Bayjuayj%5B",
            "width": 16,
          },
        ]
      }
      source={
        [
          {
            "uri": "https://example.com/image.jpg",
          },
        ]
      }
      style={
        {
          "borderRadius": 10,
          "height": 120,
          "width": 120,
        }
      }
      transition={
        {
          "duration": 1000,
        }
      }
      width={120}
    />
    <View
      style={
        {
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingTop": 14,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 14,
              "lineHeight": 16,
            },
            {
              "fontFamily": "Montserrat-Medium",
              "fontWeight": "500",
            },
            {
              "color": "#4f4f4f",
            },
            undefined,
          ]
        }
      >
        Test Product
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
            "gap": 14,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 18,
                "lineHeight": 18,
              },
              {
                "fontFamily": "Montserrat-SemiBold",
                "fontWeight": "700",
              },
              {
                "color": "#33907c",
              },
              undefined,
            ]
          }
        >
          $
          90
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 5,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "#4f4f4f",
                },
                {
                  "textDecorationLine": "line-through",
                },
              ]
            }
          >
            $
            100
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 14,
                  "lineHeight": 16,
                },
                {
                  "fontFamily": "Montserrat-Medium",
                  "fontWeight": "500",
                },
                {
                  "color": "#4f4f4f",
                },
                undefined,
              ]
            }
          >
            10
            % off
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 12,
                "lineHeight": 21,
              },
              {
                "fontFamily": "Montserrat-Medium",
                "fontWeight": "500",
              },
              {
                "color": "#4f4f4f",
              },
              undefined,
            ]
          }
        >
          Qty:
        </Text>
        <View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                },
                undefined,
              ]
            }
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
              testID="ios_touchable_wrapper"
            >
              <View
                pointerEvents="box-only"
              >
                <TextInput
                  editable={false}
                  style={
                    [
                      undefined,
                      {},
                    ]
                  }
                  testID="text_input"
                  value="2"
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
